src/main/java/org/aion/wallet/ui/components/partials/ImportAccountDialog.java

```diff
@@ -1,71 +1,136 @@
-        AccountDTO account = blockchainConnector.addKeystoreUTCFile(Files.readAllBytes(file.toPath()), password);
-        EventPublisher.fireAccountChanged(account);
+        File file = fileChooser.showOpenDialog(null);
+        if (file == null) {
+            return;
+        }
+        keystoreTextView.setText(file.getName());
+        keystorePassword.requestFocus();
+        keystoreFile = Files.readAllBytes(file.toPath());
     }
 
-    public void uploadPrivateKey() {
-
+    public void createAccount(MouseEvent mouseEvent) throws ValidationException {
+        if (importKeystoreView.isVisible()) {
+            //todo: add validation
+            String password = keystorePassword.getText();
+            AccountDTO account = blockchainConnector.addKeystoreUTCFile(keystoreFile, password);
+            EventPublisher.fireAccountChanged(account);
+        } else {
+            //todo: import private key
+        }
+        this.close(mouseEvent);
     }
 
     public void open(MouseEvent mouseEvent) {
-        popup.setAutoHide(true);
-        popup.setAutoFix(true);
-
+        StackPane pane = new StackPane();
         Pane importAccountDialog = null;
         try {
             importAccountDialog = FXMLLoader.load(getClass().getResource("ImportAccountDialog.fxml"));
         } catch (IOException e) {
             e.printStackTrace();
             return;
         }
-        popup.getScene().getWindow().centerOnScreen();
+        pane.getChildren().add(importAccountDialog);
+        Scene secondScene = new Scene(pane, importAccountDialog.getPrefWidth(), importAccountDialog.getPrefHeight());
+        secondScene.setFill(Color.TRANSPARENT);
+
+        Stage popup = new Stage();
+        popup.setTitle("Import account");
+        popup.setScene(secondScene);
 
         Node eventSource = (Node) mouseEvent.getSource();
         popup.setX(eventSource.getScene().getWindow().getX() + eventSource.getScene().getWidth() / 2 - importAccountDialog.getPrefWidth() / 2);
         popup.setY(eventSource.getScene().getWindow().getY() + eventSource.getScene().getHeight() / 2 - importAccountDialog.getPrefHeight() / 2);
-        popup.getContent().addAll(importAccountDialog);
-        popup.show(eventSource.getScene().getWindow());
+        popup.initModality(Modality.APPLICATION_MODAL);
+        popup.initStyle(StageStyle.TRANSPARENT);
+
+        popup.show();
     }
 
-    public void close() {
-        popup.hide();
+    public void close(MouseEvent mouseEvent) {
+        ((Node) mouseEvent.getSource()).getScene().getWindow().hide();
     }
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        privateKeyRadioButton.setUserData(PK_RADIO_BUTTON_ID);
+        keystoreRadioButton.setUserData(KEYSTORE_RADIO_BUTTON_ID);
+        accountTypeToggleGroup.selectedToggleProperty().addListener((ObservableValue<? extends Toggle> ov, Toggle old_toggle, Toggle new_toggle) -> {
```

Comment:
you could extract this to a reference method eg: `(this:radioButtonChanged)` 
and then use the new_toggle value to switch instead of `accountTypeToggleGroup.getSelectedToggle()`

Also, names should be`newToggle` instead of `new_toggle` etc.